doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,62 @@
+=======================
+ MEP 28 Artist Refactor
+=======================
+
+.. contents::
+   :local:
+
+
+Branches and Pull requests
+==========================
+
+All development branches containing work on this MEP should be linked to from here.
+
+All pull requests submitted relating to this MEP should be linked to
+from here.  (A MEP does not need to be implemented in a single pull
+request if it makes sense to implement it in discrete phases).
+
+Abstract
+========
+
+During the course of MPL, the number and complexity of artists have grown, to become quite unmanageable, this PR seeks to address that by refactoring the Artists and creating a more sensible class structure.
+
+
+Detailed description
+====================
+
+The goal of this MEP lies in automating and reducing as much of the "admin" code as possible making it easier to see what goes on and where and make faster changes to the codebase.  A lot of the artist code for the example gets taken up with simple getter/setters, very laborious, and when it comes to the documentation it looks like a getter/setter desert. More so, we the same properties duplicated throughout the codebase, such as fillcolor, where we have ``set_fillcolor`` methods in ``lines``, ``markers`` and ``patches``.  This makes readability and maintainability very difficult and just adds to the Artist doc desert.
+
+With a clearer structure it also makes it easier to add new features such as keeping the legend upto date with its parent artists, add make it easier to implement dynamical property changing.  
+
+
+Implementation
+==============
+
+## Step 1 - Remove getter/setters
+The easiest and biggest reduction comes from removing all of the ``artist.set_xxx(value)`` methods (where possible).
```

Comment:
By using plain attributes, or by using properties?  With the latter, we will still have getters and setters, but not as part of the API.